backends/ebpf/psa/ebpfPsaTable.cpp

```diff
@@ -0,0 +1,288 @@
+                           CodeGenInspector* codeGen, cstring name, size_t size) :
+                           EBPFTable(program, table, codeGen), name(name), size(size) {
+    auto sizeProperty = table->container->properties->getProperty("size");
+    if (keyGenerator == nullptr && sizeProperty != nullptr) {
+        ::warning(ErrorType::WARN_IGNORE_PROPERTY,
+                  "%1%: property ignored due to not defined table key", sizeProperty);
+    }
+    if (keyFieldNames.empty() && size != 1) {
+        if (sizeProperty != nullptr) {
+            ::warning(ErrorType::WARN_IGNORE,
+                      "%1%: only one entry allowed with empty key or selector-only key",
+                      sizeProperty);
+        }
+        this->size = 1;
+    }
+}
+
+
+void EBPFTablePSA::emitValueActionIDNames(CodeBuilder* builder) {
+    // For action_run method we preserve these ID names for actions.
+    // Values are the same as for implementation, because the same action
+    // set is enforced.
+    if (singleActionRun()) {
+        EBPFTable::emitValueActionIDNames(builder);
+    }
+}
+
+void EBPFTablePSA::emitValueStructStructure(CodeBuilder* builder) {
+    // TODO: placeholder for handling psa_implementation
+    EBPFTable::emitValueStructStructure(builder);
+}
+
+void EBPFTablePSA::emitInstance(CodeBuilder *builder) {
+    if (keyGenerator != nullptr) {
+        TableKind kind = isLPMTable() ? TableLPMTrie : TableHash;
+        emitTableDecl(builder, name, kind,
+                      cstring("struct ") + keyTypeName,
+                      cstring("struct ") + valueTypeName, size);
+    }
+
+    emitTableDecl(builder, defaultActionMapName, TableArray,
+                  program->arrayIndexType,
+                  cstring("struct ") + valueTypeName, 1);
+}
+
+void EBPFTablePSA::emitTableDecl(CodeBuilder *builder,
+                                 cstring tblName,
+                                 TableKind kind,
+                                 cstring keyTypeName,
+                                 cstring valueTypeName,
+                                 size_t size) const {
+    builder->target->emitTableDecl(builder,
+                                   tblName, kind,
+                                   keyTypeName,
+                                   valueTypeName,
+                                   size);
+}
+
+void EBPFTablePSA::emitTypes(CodeBuilder* builder) {
+    EBPFTable::emitTypes(builder);
+    // TODO: placeholder for handling PSA-specific types
+}
+
+void EBPFTablePSA::emitAction(CodeBuilder* builder, cstring valueName, cstring actionRunVariable) {
+    // TODO: placeholder for handling psa_implementation
+    EBPFTable::emitAction(builder, valueName, actionRunVariable);
+}
+
+void EBPFTablePSA::emitInitializer(CodeBuilder *builder) {
+    this->emitDefaultActionInitializer(builder);
+    this->emitConstEntriesInitializer(builder);
+}
+
+void EBPFTablePSA::emitConstEntriesInitializer(CodeBuilder *builder) {
```

Comment:
Actually, I believe the classic eBPF backend has already support for const entries: 

https://github.com/p4lang/p4c/blob/main/backends/ebpf/ebpfTable.cpp#L566. 

However, there is a difference how the control-plane API look like for these two architectures (e.g., `ebpf_model` uses `BPF_OBJ_GET`), so that it cannot be re-used.

Yes, it's executed by the control plane. See Control-plane API section from the updated README.